src/compiler/scala/tools/nsc/tasty/bridge/TypeOps.scala

```diff
@@ -418,7 +418,7 @@ trait TypeOps { self: TastyUniverse =>
         if (self.typeSymbol.isSubClass(from)) {
           def elemType(tp: Type): Type = tp.dealiasWiden match {
             // case tp: AndOrType => tp.derivedAndOrType(elemType(tp.tp1), elemType(tp.tp2))
-            case tp: u.RefinedType => u.intersectionType(tp.parents.map(elemType))
+            case tdw: u.RefinedType => u.intersectionType(tdw.parents.map(elemType))
```

Comment:
sample where `q.m match { case q: Q => }` could be excluded, like `x match { case x: X => }`.